### ISSUE TITLE
Adding query abort functionality

### ISSUE THREAD

**Issue by thehacklady** [ToolJet/ToolJet#7149]
Currently when a user miss to add limit to SQL query and clicks preview - it fetch all the 1L+ rows (lets say) in the table(taking it as example), which is consuming resources. They need to refresh the app, to halt the request.

Improvement - We should have a functionality to abort queries.

**Comment by webAbhi** [comment]
@thehacklady can you please add the step by step process to test this, maybe attaching some screenshot in this issue would help a lot !! this issue looks really interesting to me.

**Comment by tejakummarikuntla** [comment]
Hey @webAbhi, Thank you for expressing your interest in contributing to our project. However, I must inform you that the particular issue you're referring to is not labeled as "up for grab." As a result, we won't be able to assign this specific issue at the moment.

Nevertheless, I encourage you to explore other issues that are labeled as "up for grabs." and it would be a great way to contribute to the project. Thank you for your understanding and willingness to contribute to the project

**Comment by Kartik18g** [comment]
We'll also push the abort log into the debugger now.

**Comment by overlyhonest** [comment]
[updated design](https://www.figma.com/design/98yBvWcXJ6Obn8swxk0zQi/Query-editor-2.0?node-id=5135-201912&t=Z3GkYxkhBEVmuK5z-11)

**Comment by johnsoncherian** [comment]
### Proposed Implementation

Introduce a **client-side query abort mechanism** to allow users and developers to stop waiting for long-running queries.

### 1. Programmatic Abort API

Expose an abort interface:

```js id="r9u1ph"
queries.${queryName}.abort()
```

* Internally uses **AbortController** (or equivalent)
* Cancels the in-flight HTTP request from the client
* Immediately frees up the UI from waiting on the response

### 2. Behavior and Scope

* Aborting a query:

  * Terminates the request on the **client side**
  * Stops further response handling in the application

### 3. Important Limitation (Explicit)

This is a **client-side abort only**:

* The backend will continue processing the request
* The database query will continue execution until completion
* No compute resources are reclaimed at backend/DB level

In other words:

* Abort = "stop waiting", not "stop execution"
